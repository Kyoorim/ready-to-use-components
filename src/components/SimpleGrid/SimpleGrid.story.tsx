import React, { FC } from 'react';
import { Meta } from '@storybook/react';
import SimpleGrid, { SimpleGridProps } from './SimpleGrid';
import Box from '../Box/Box';

export default {
  title: 'SimpleGrid',
  component: SimpleGrid,
} as Meta<SimpleGridProps>;

export const Default: FC = () => {
  return (
    <SimpleGrid>
      <Box>1</Box>
      <Box>2</Box>
      <Box>3</Box>
      <Box>4</Box>
      <Box>5</Box>
      <Box>6</Box>
    </SimpleGrid>
  );
};
