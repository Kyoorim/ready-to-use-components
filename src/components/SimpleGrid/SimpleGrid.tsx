import React from 'react';
import styled from 'styled-components';
import { flexbox, FlexboxProps, grid, GridProps, layout, LayoutProps, space, SpaceProps } from 'styled-system';

const SimpleGridBox = styled.div<CustomProps>`
  ${layout}
  ${space}
  ${flexbox}
  ${grid}
  display: grid;
  grid-column-gap: ${(props) => `${props.spacingX}px`};
  grid-row-gap: ${(props) => `${props.spacingY}px`};
  grid-template-columns: ${(props) => `repeat(${props.column}, 1fr)`};
  grid-template-rows: ${(props) => `repeat(${props.row}, 1fr)`};
`;

export interface CustomProps {
  column?: number;
  row?: number;
  spacingX?: string | number;
  spacingY?: string | number;
}

export type SimpleGridProps = CustomProps & LayoutProps & SpaceProps & FlexboxProps & GridProps;

const SimpleGrid: FCC<SimpleGridProps> = ({ column = 3, row = 3, spacingX = '20px', spacingY = '20px', ...props }) => {
  return (
    <SimpleGridBox
      data-testid="simpleGrid-comp"
      column={column}
      row={row}
      spacingX={spacingX}
      spacingY={spacingY}
      {...props}
    ></SimpleGridBox>
  );
};

export default SimpleGrid;
