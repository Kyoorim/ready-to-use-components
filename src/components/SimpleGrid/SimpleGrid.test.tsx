import React from 'react';
import { render, screen } from '@testing-library/react';
import SimpleGrid from './SimpleGrid';

const defaultProps = {
  onClick: jest.fn(),
  children: 'SimpleGrid',
};

describe('<SimpleGrid />', () => {
  it('should render', () => {
    render(<SimpleGrid {...defaultProps} />);

    expect(screen.getByTestId('simpleGrid-comp')).toBeInTheDocument();
  });

  it('should render children', () => {
    render(<SimpleGrid {...defaultProps} />);

    expect(screen.getByText(/SimpleGrid/i)).toBeInTheDocument();
  });

  it('should allow custom className', () => {
    const props = {
      ...defaultProps,
      className: 'Custom',
    };
    render(<SimpleGrid column={3} {...props} />);

    expect(screen.getByTestId('simpleGrid-comp')).toHaveProperty('className');
  });

  it('should render items under grid style', () => {
    render(<SimpleGrid {...defaultProps} />);

    expect(screen.getByTestId('simpleGrid-comp')).toHaveStyle('display: grid');
  });
});
