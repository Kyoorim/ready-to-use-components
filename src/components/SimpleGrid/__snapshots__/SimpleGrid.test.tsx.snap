// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleGrid /> should render 1`] = `
<div
  class="sc-bcXHqe lkQTgH"
  data-testid="simpleGrid-comp"
>
  SimpleGrid
</div>
`;
